src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -153,6 +153,9 @@ internal void EnsureInitialized(WebHostSettings settings)
                             // settings configured by the user (WEBSITE_TIME_ZONE) are honored.
                             // DateTime caches timezone information, so we need to clear the cache.
                             TimeZoneInfo.ClearCachedData();
+
+                            // ensure we reinitialize hostname after specialization
+                            HostNameProvider.Reset();
```

Comment:
I actually need to check the v2 implementation and ensure that the HostNameService is recreated on specialization